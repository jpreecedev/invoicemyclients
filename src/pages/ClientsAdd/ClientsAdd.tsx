import React from 'react'

import { AuthenticatedPage } from '../../components/AuthenticatedPage'
import { Container } from '../../components/Container'
import { ClientDetails } from '../../components/ClientDetails'

const ClientsAddPage = () => {
  return (
    <AuthenticatedPage title="Add New Client">
      <Container additionalClasses="col-12 col-md-6">
        <h2>Details</h2>
        <ClientDetails />
      </Container>
      <Container additionalClasses="col-12 col-md-6">
        <h2>Details</h2>
        <ClientDetails />
      </Container>
    </AuthenticatedPage>
  )
}

export { ClientsAddPage }
