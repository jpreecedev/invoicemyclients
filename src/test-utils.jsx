import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
