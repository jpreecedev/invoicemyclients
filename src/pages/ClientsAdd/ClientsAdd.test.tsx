import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { ClientsAddPage } from './ClientsAdd'

describe('<ClientsAdd /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<ClientsAddPage />).container).toMatchSnapshot()
  })
})
