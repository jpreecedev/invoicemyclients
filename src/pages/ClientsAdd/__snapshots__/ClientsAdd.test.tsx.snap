// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientsAdd /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <main
    class="mt-5 mb-10 pb-10"
  >
    <h1
      class="container"
    >
      <div
        class="row"
      >
        <h1>
          Add New Client
        </h1>
      </div>
      <div
        class="row"
      >
        <section
          class="container shadow-sm mb-5 rounded-lg col-12 col-md-6"
        >
          <h2>
            Details
          </h2>
          <form>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="companyName"
              >
                <small>
                  Company Name
                </small>
              </label>
              <input
                class="form-control"
                id="companyName"
                placeholder="Company Name"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="idNumber"
              >
                <small>
                  ID Number
                </small>
              </label>
              <input
                class="form-control"
                id="idNumber"
                placeholder="ID Number"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="website"
              >
                <small>
                  Website
                </small>
              </label>
              <input
                class="form-control"
                id="website"
                placeholder="Website"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="vatNumber"
              >
                <small>
                  VAT Number
                </small>
              </label>
              <input
                class="form-control"
                id="vatNumber"
                placeholder="VAT Number"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="phoneNumber"
              >
                <small>
                  Phone Number
                </small>
              </label>
              <input
                class="form-control"
                id="phoneNumber"
                placeholder="Phone Number"
                type="text"
              />
            </div>
          </form>
        </section>
        <section
          class="container shadow-sm mb-5 rounded-lg col-12 col-md-6"
        >
          <h2>
            Details
          </h2>
          <form>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="companyName"
              >
                <small>
                  Company Name
                </small>
              </label>
              <input
                class="form-control"
                id="companyName"
                placeholder="Company Name"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="idNumber"
              >
                <small>
                  ID Number
                </small>
              </label>
              <input
                class="form-control"
                id="idNumber"
                placeholder="ID Number"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="website"
              >
                <small>
                  Website
                </small>
              </label>
              <input
                class="form-control"
                id="website"
                placeholder="Website"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="vatNumber"
              >
                <small>
                  VAT Number
                </small>
              </label>
              <input
                class="form-control"
                id="vatNumber"
                placeholder="VAT Number"
                type="text"
              />
            </div>
            <div
              class="form-group narrow d-flex"
            >
              <label
                class="w-25 d-flex align-items-center"
                for="phoneNumber"
              >
                <small>
                  Phone Number
                </small>
              </label>
              <input
                class="form-control"
                id="phoneNumber"
                placeholder="Phone Number"
                type="text"
              />
            </div>
          </form>
        </section>
      </div>
    </h1>
  </main>
</div>
`;
